scripts/solve.js

```diff
@@ -6,14 +6,29 @@ window.addEventListener('load', function() {
     mode: 'javascript',
     lineNumbers: true
   });
-
+  
   runStaticAnalysis(myCodeMirror.getValue());
+  
+  myCodeMirror.on('change', function() {
+    runStaticAnalysis(myCodeMirror.getValue());
+  });
 });
 
 function runStaticAnalysis(code) {
```

Comment:
this function looks like it will grow to be large and complex in the future. We should consider a refactor after the prototype.

I propose tracking these work items either in Github or Buganizer and then leaving a TODO in the code with the URL to the tracking issue.

Reply:
I just rebased to pull in Nick's commits -- does that address the concern or should I still add a TODO?

Reply:
lgtm